.github/dependabot.yml

```diff
@@ -0,0 +1,8 @@
+version: 2
+updates:
+
+  # Check for updates to GitHub Actions
```

Comment:
Would it make sense to group some of these into a single PR? Seems a bit noisy given how many different actions we use: https://github.com/marbre/iree/pulls

![image](https://github.com/user-attachments/assets/dfe014e1-ef12-454b-b8a5-d3589e3aafe8)

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/optimizing-pr-creation-version-updates#prioritizing-meaningful-updates

Reply:
Fair enough. I iterate on it.

Reply:
Splitting into between what lives in `actions/*` and "rest of the world" could be a good start. WDYT? I can send a follow up if that makes sense to you.

Reply:
Let's live with this for a bit and then split as needed.